R/psprintf.R

```diff
@@ -0,0 +1,36 @@
+#' Paste together sprintf-style text and variable values
+#'
+#' Helper function to sprintf a bunch of key-value (string-variableVector) pairs,
+#' then paste them together with a good separator for contructing remake recipes.
+#'
+#' @param ... list of sprintf arguments seperated by commas
+#' @param sep separator for the list of arguments, defaults to new line
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' psprintf(
+#'   "plot_frame(",
+#'   "png_file = target_name,",
+#'   "date = date_%s," = "2018-01-23",
+#'   "state = %s)" = "Wisconsin")
+#' }
```

Comment:
The above is a good example, and maybe the best one, but for the record (and if you think these should go in, please add them), here are a couple of others that exercise the functionality from my `...` description above:
```
cat(psprintf(
  'cmd(',
  'a = %s, b = \'%s\','=c('dog','cat'),
  'c = "rat")'
))
## cmd(
##       a = dog, b = 'cat',
##       c = "rat")
```

```
cat(psprintf(
  'cmd(',
  'a = %s, b = %0.1f,'=list('dog',7),
  'c = "rat")'
))
## cmd(
##       a = dog, b = 7.0,
##       c = "rat")
```

Reply:
I will add those examples. Thanks!